src/test/java/org/apache/sysds/test/functions/builtin/part1/BuiltinImageCutoutLinTest.java

```diff
@@ -0,0 +1,133 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.sysds.test.functions.builtin.part1;
+
+import org.apache.sysds.common.Types.ExecMode;
+import org.apache.sysds.common.Types.ExecType;
+import org.apache.sysds.runtime.matrix.data.MatrixValue;
+import org.apache.sysds.test.AutomatedTestBase;
+import org.apache.sysds.test.TestConfiguration;
+import org.apache.sysds.test.TestUtils;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Random;
+// if A should be generated one row at a time
+//import java.util.stream.Stream;
+//import java.util.stream.DoubleStream;
+
+public class BuiltinImageCutoutLinTest extends AutomatedTestBase {
+    private final static String TEST_NAME = "image_cutout_linearized";
+    private final static String TEST_DIR = "functions/builtin/";
+    private final static String TEST_CLASS_DIR = TEST_DIR + BuiltinImageCutoutLinTest.class.getSimpleName() + "/";
+
+    private final static double eps = 1e-10;
+    private final static double spSparse = 0.1;
+    private final static double spDense = 0.9;
+    private final static Random random = new Random();
+
+    @Override
+    public void setUp() {
+        addTestConfiguration(TEST_NAME, new TestConfiguration(TEST_CLASS_DIR, TEST_NAME, new String[] { "B" }));
+    }
+
+    @Test
+    public void testImageTranslateMatrixDenseCP() {
+        runImageCutoutLinTest(false, ExecType.CP);
+    }
+
+    @Test
+    public void testImageTranslateMatrixSparseCP() {
+        runImageCutoutLinTest(true, ExecType.CP);
+    }
+
+    @Test
+    public void testImageTranslateMatrixDenseSP() {
+        runImageCutoutLinTest(false, ExecType.SPARK);
+    }
+
+    @Test
+    public void testImageTranslateMatrixSparseSP() {
+        runImageCutoutLinTest(false, ExecType.SPARK);
+    }
+
+    private void runImageCutoutLinTest(boolean sparse, ExecType instType) {
+        ExecMode platformOld = setExecMode(instType);
+        disableOutAndExpectedDeletion();
+
+        setOutputBuffering(true);
+
+        int s_rows = random.nextInt(100) + 1;
+        int s_cols = random.nextInt(100) + 1;
+        int x = random.nextInt(s_cols);
+        int y = random.nextInt(s_rows);
+        int width = random.nextInt(s_cols - x) + 1;
+        int height = random.nextInt(s_rows - y) + 1;
+        int fill_color = random.nextInt(256);
+        int n_imgs = random.nextInt(100) + 1;
```

Comment:
this random behavior is not good, since each test then behaves randomly and we cannot reproduce.

Use a parameterized class instead like in PR #1903